test/hooks.on-listen.test.js

```diff
@@ -0,0 +1,614 @@
+'use strict'
+
+const { t, before } = require('tap')
+const Fastify = require('../fastify')
+const fp = require('fastify-plugin')
+const dns = require('dns').promises
+
+let localhost
+
+before(async function () {
+  const lookup = await dns.lookup('localhost')
+  localhost = lookup.address
+})
+
+t.test('localhost onListen should be called in order', t => {
+  t.plan(2)
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+  let order = 0
+
+  fastify.addHook('onListen', function (done) {
+    t.equal(order++, 0, '1st called in root')
+    done()
+  })
+
+  fastify.addHook('onListen', function (done) {
+    t.equal(order++, 1, '2nd called in root')
+    done()
+  })
+  fastify.listen({
+    host: 'localhost',
+    port: 0
+  })
+})
+t.test('localhost async onListen should be called in order', async t => {
```

Comment:
nitpick:
Some newlines would make it less textwall. Following cases should have also newlines.

```suggestion
})

t.test('localhost async onListen should be called in order', async t => {
```